lib/AST/Attr.cpp

```diff
@@ -1376,7 +1376,8 @@ bool DeclAttribute::printImpl(ASTPrinter &Printer, const PrintOptions &Options,
             if (macroIntroducedNameRequiresArgument(name.getKind())) {
               SmallString<32> buffer;
               StringRef nameText = name.getName().getString(buffer);
-              bool shouldEscape = nameText == "$";
+              bool shouldEscape = nameText == "$" || nameText == "init" ||
```

Comment:
Maybe it would be better for me to just restore the call to `escapeKeywordInContext()` here? That was removed in https://github.com/apple/swift/pull/65362 but I assume that's just because the parsing was upgraded in that PR.

Reply:
`escapeKeywordInContext` is probably the right thing whenever the base name is an identifier---I never should have removed it in the first place---but I'm happy with this as workaround.

Reply:
We've got time on this one, I can put in the more appropriate fix.